### ISSUE TITLE
Selecting "AutoFill Passwords Settings" doesn't take me to the proper page

### ISSUE THREAD

**Issue by past** [mozilla-lockbox/lockbox-ios#747]
In Lockbox 1.3 (2340) selecting "AutoFill Passwords Settings" takes me to the Settings app, but doesn't open the AutoFill Passwords page. If I kill the Settings app prior to this action, it just takes me to the Settings front page. Otherwise it will just remain on whatever page was last visited in the Settings app.

**Comment by sandysage** [comment]
@changecourse Yes, let's prioritize this design and implementation work. I'm marking this as a P2 with a few acceptance criteria:
* Should inform the user _how_ to set up autofill in device settings
* Should have call to action that directs the user to device settings
* Should have a way to get back to this instruction within the app for all the times users dismiss instructions and need to return
* Should display for all users - new and returning

Question: @devinreams can we have it so that a link to device settings within our app always takes the user to the "general" (root) screen? or does it have to go to settings where the user last left off?

**Comment by sandysage** [comment]
My selfish goal here is to also achieve the aims laid out in #733 for existing users.

**Comment by changecourse** [comment]
A proposed design approach for the three separate use cases for needing instructions:
1. existing Lockbox users, upon receiving the Autofill update for the first time
2. new Lockbox users, as part of the onboarding experience, after FxA, but before the confirmation screen
3. existing Lockbox users, who skip autofill setup and choose to access it later via Settings

_**This approach also assumes that we can detect when autofill has been enabled, thus dismissing the instructions screens accordingly upon successful completion**_


### Visual Design

**InVision**
Onboarding / Update Promo: https://mozilla.invisionapp.com/share/BQEUSO0S2#/316626649_AutoFill_-_Onboarding
Onboarding / Update Instructions: https://mozilla.invisionapp.com/share/BQEUSO0S2#/320929937_AutoFill_-_Instruction
Settings Instructions: https://mozilla.invisionapp.com/share/BQEUSO0S2#/320929936_Settings_-_Set_Up_AutoFill

**Zeplin**
Onboarding / Update Promo: https://zpl.io/bLDEeWY
Onboarding / Update Instructions: https://zpl.io/brEq9j5
Settings Instructions: https://zpl.io/b6vnGOK

**Comment by changecourse** [comment]
@devinreams & @sashei can you review at some point and we can discuss over standup as needed this week or Monday, ahead of the afternoon iOS meeting?

**Comment by sashei** [comment]
@sandysage there is not a way to deep link to the "root" settings screen without using a private API.

**Comment by devinreams** [comment]
>>Question: @devinreams can we have it so that a link to device settings within our app always takes the user to the "general" (root) screen? or does it have to go to settings where the user last left off?

>there is not a way to deep link to the "root" settings screen without using a private API.

To reiterate: the linking that happens now goes directly into the app-specific's settings page. (Settings > Lockbox) and not the top level Settings root view, nor where the user left off. If you experience that behavior its unexpected.

---

Another open question was: can the app can tell if it's enabled as an AutoFill provider?

I looked around and don't believe we can.. @sashei do you know of any way?

I can file some radars with Apple to suggest that, in addition to putting the AutoFill provider/permission on the app-specific settings screen...

**Comment by nickbrandt** [comment]
@devinreams in regards to the second question (knowing if Lockbox is enabled as an AutoFill provider), we were assuming that was possible since there is logic in place to not show the onboarding screen currently if Lockbox is already enabled (https://github.com/mozilla-lockbox/lockbox-ios/pull/679#pullrequestreview-152765669). Was unsure though if that check could be made on foregrounding the Lockbox app after going to Settings.

**Comment by sashei** [comment]
@devinreams @nickbrandt it is possible to tell if the app is an autofill service provider.

this can be done anytime, including when the app is foregrounded.

**Comment by sashei** [comment]
it would be cool if that was on the app-specific settings screens, we should totally file a radar for that.

**Comment by devinreams** [comment]
>>Was unsure though if that check could be made on foregrounding the Lockbox app after going to Settings.

>it is possible to tell if the app is an autofill service provider.

>this can be done anytime, including when the app is foregrounded.

Thanks for confirming 👍 

---

>it would be cool if that was on the app-specific settings screens, we should totally file a radar for that.

I submitted my own bug under ID: 44644067

**Comment by changecourse** [comment]
Updated the designs in the description above, based on our limitations with Settings. This will simplify our approach for the three use cases, so we don't have to design slightly different screens, and it will remove routing callouts, in favor of a more static set of instructions in all instances. 

Considering that there isn't an effective and consistent way to route everyone to the root of settings, only the app-specific settings, we would like to pose doing an animated gif of the steps as part of our instruction screen instead, to be able to support the 5-6 step flow required for AutoFill setup. 

Does this make more sense as an approach then, @devinreams / @sashei ?

**Comment by sashei** [comment]
@changecourse looks good to me! presumably if that radar gets addressed we will go back to linking to the app-specific settings but I think this is a great workaround for now.

**Comment by joeyg** [comment]
@changecourse I think the link for to invision "Accessing AutoFill Instructions via Settings: " needs to be updated - it requires log in.

**Comment by changecourse** [comment]
@joeyg updated. thanks for catching that!

**Comment by sashei** [comment]
I suspect that we will run into scaling issues with the above approach. I asked the iOS team for some recommendations here; some options that came out of that conversation are [Lottie](https://github.com/airbnb/lottie-ios) and using the .mp4 file format with all video controls hidden on-screen.

implementer's choice about what's best here, but there's a few!!

**Comment by changecourse** [comment]
I love Lottie, but am not sure that it would be a good candidate for this workflow, spanning multiple screens, since we'd need to build all of the individual UI elements and animations for screens manually, rather than rely on the screencast of the actual flow... 

I'm curious about using an mp4 vs. the .mov (apple quicktime format) which was provided above... I'll do some more digging on codec approach; my initial attempt at exporting an mp4 was not successful, due to compression issues and having difficulty maintaining the custom aspect ratio for this crop.

**Comment by changecourse** [comment]
Actually, might have figured out a good approach for an .mp4, if that is preferable to a .mov

[AutofillSetup_v1.4.mp4.zip](https://github.com/mozilla-lockbox/lockbox-ios/files/2425547/AutofillSetup_v1.4.mp4.zip)
